lib/request_handler/option_parser.rb

```diff
@@ -12,7 +12,7 @@ def initialize(params:, allowed_options_type:)
     private
 
     def enum?
```

Comment:
doesn't this now change what this method is doing? before it checked for a enum, now it just checks that the schema is valid

Reply:
yes a bit but I thought as long as it can validate something it should be fine to use? If we specifically want to check for an enum we'll need one check per backend

Reply:
This is change that needs to be documented then. Also this might cause some edge cases which where avoided by only allowing enums before.

Reply:
As long as you pass an enum as in the docs, nothing changes.  The only problem I see is that it would maybe not cause an informative error message right away if you use something that allows hashes or something for example

Reply:
please reflect the change at least in the method name - it no longer checks if the type is a enum type